mdata/aggmetric.go

```diff
@@ -391,7 +389,7 @@ func (a *AggMetric) persist(pos int) {
 		previousPos += len(a.chunks)
 	}
 	previousChunk := a.chunks[previousPos]
-	for (previousChunk.Series.T0 < chunk.Series.T0) && (a.lastSaveStart < previousChunk.Series.T0) {
+	for (previousChunk.Series.T0 < chunk.Series.T0) && (lastSaveStart < previousChunk.Series.T0) {
```

Comment:
it is possible that:
1) during runtime of this function, more chunk saves synced their state and updated lastSaveStart.
2) we work with an outdated lastSaveStart here, and hence, may re-save those same chunks.

But this is similar to before (with current master, we would also resave them, as the SyncChunkSaveState calls would be blocked from executing while this function runs). so just an observation.

Reply:
we could just atomically load it again here. i wasn't sure about which side of the trade-off is preferable when comparing between introducing the over head of another call to `atomic.Load` compared to accepting that there is a very unlikely scenario in which we double save a chunk.